relay-server/src/processing/spans/process.rs

```diff
@@ -84,3 +85,472 @@ fn normalize_span(
+}
+
+#[cfg(test)]
+mod tests {
+    use relay_pii::{DataScrubbingConfig, PiiConfig};
+    use relay_protocol::SerializableAnnotated;
+
+    use crate::services::projects::project::ProjectInfo;
+
+    use super::*;
+
+    fn make_context(
+        scrubbing_config: DataScrubbingConfig,
+        pii_config: Option<PiiConfig>,
+    ) -> Context<'static> {
+        let config = Box::leak(Box::new(relay_config::Config::default()));
+        let global_config = Box::leak(Box::new(relay_dynamic_config::GlobalConfig::default()));
+        let project_info = Box::leak(Box::new(ProjectInfo {
+            config: relay_dynamic_config::ProjectConfig {
+                pii_config,
+                datascrubbing_settings: scrubbing_config,
+                ..Default::default()
+            },
+            ..Default::default()
+        }));
+        let rate_limits = Box::leak(Box::new(relay_quotas::RateLimits::default()));
+
+        Context {
+            config,
+            global_config,
+            project_info,
+            rate_limits,
+            sampling_project_info: None,
+        }
+    }
+
+    #[test]
+    fn test_scrub_span_pii_default_rules_links() {
+        // `user.name`, `sentry.release`, and `url.path` are marked as follows in `sentry-conventions`:
+        // * `user.name`: `true`
+        // * `sentry.release`: `false`
+        // * `url.path`: `maybe`
+        // Therefore, `user.name` is the only one that should be scrubbed by default rules.
+        let json = r#"{
+            "start_timestamp": 1544719859.0,
+            "end_timestamp": 1544719860.0,
+            "trace_id": "5b8efff798038103d269b633813fc60c",
+            "span_id": "eee19b7ec3c1b174",
+            "name": "test",
+            "links": [{
+                "trace_id": "5b8efff798038103d269b633813fc60c",
+                "span_id": "eee19b7ec3c1b174",
+                "attributes": {
+                    "sentry.description": {
+                        "type": "string",
+                        "value": "secret123"
+                    },
+                    "user.name": {
+                        "type": "string",
+                        "value": "secret123"
+                    },
+                    "sentry.release": {
+                        "type": "string",
+                        "value": "secret123"
+                    }
+                }
+            }]
+        }"#;
+
+        let mut data = Annotated::<SpanV2>::from_json(json).unwrap();
+
+        let mut scrubbing_config = relay_pii::DataScrubbingConfig::default();
+        scrubbing_config.scrub_data = true;
+        scrubbing_config.scrub_defaults = true;
+        scrubbing_config.scrub_ip_addresses = true;
+        scrubbing_config.sensitive_fields = vec![
+            "value".to_owned(), // Make sure the inner 'value' of the attribute object isn't scrubbed.
+            "very_sensitive_data".to_owned(),
+        ];
+        scrubbing_config.exclude_fields = vec!["public_data".to_owned()];
```

Comment:
Elsewhere we use the struct initializer `DataScrubingConfig { foo: bar, ..Default::default() } `



```suggestion
        let scrubbing_config = relay_pii::DataScrubbingConfig {
            scrub_data = true,
            scrub_defaults = true,
            scrub_ip_addresses = true,
            sensitive_fields = vec![
                "value".to_owned(), // Make sure the inner 'value' of the attribute object isn't scrubbed.
                "very_sensitive_data".to_owned(),
            ],
            exclude_fields = vec!["public_data".to_owned()],
            ..Default::default()
        }
```

Reply:
Yeah, `clippy` made me do that in the `relay-pii` crate. The problem is that the `pii_config` field is accessible there but not in this crate, so we can initialize the struct directly in `relay-pii` but not in `relay-server`.

Reply:
Ah right, that's annoying 👍